Mem object

qc-azarrabi opened the pull request:
All commit to prepare for memory object, including:
  - extending ns code to handle typed objects including changes to synchronization as ns table users extended to local users.
  - differentiate between local id and id seen by QTEE.
  - clean up and bugfixes.

qc-azarrabi committed:
libqcomtee: create private header

Move some shared symbols from qcomte_object.c.
Move tee_call_t refernce to the root object to keep life easier.

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

qc-azarrabi committed:
libqcomtee: cleanup

Rename some symbols for consistency.

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

qc-azarrabi committed:
libqcomtee: add support for memory object

Introduce these APIs:
  - qcomtee_memory_object_alloc to create a new memory object.
  - qcomtee_memory_object_addr to get address of memory object.
  - qcomtee_memory_object_size to get size of memory object.
  - qcomtee_memory_object_release to release the meory object.

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

qc-azarrabi committed:
libqcomtee: fix return value for id allocator

The return values was not handelled.
Also, some cleanups for namespace codes.

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

qc-azarrabi committed:
libqcomtee: add new OBJREF flags

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

qc-azarrabi committed:
libqcomtee: introduce tee_object_id

 - Add new object ID. This is the ID seen by QTEE. The original
   object_id is the ID visibale locally.
 - Improve ns table to accept type to search for tee_object_id.

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

qc-azarrabi committed:
test: fix broken compile

Update based on commit 9c20f362e846.

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

qc-azarrabi committed:
libqcomtee: fix release operation id

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

qc-azarrabi committed:
libqcomtee: improve error mrssages

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

harshaldev27 commented on libqcomtee/CMakeLists.txt:
```diff
@@ -22,6 +22,7 @@ include_directories(${QCBOR_INCLUDE_DIRS})
 set(SRC
         src/qcomtee_object.c
         src/objects/credentials_obj.c
+	src/objects/mem_obj.c
```
Indent.

qc-azarrabi replied:
Ack.

harshaldev27 commented on libqcomtee/src/objects/mem_obj.c:
```diff
@@ -0,0 +1,97 @@
+#include <qcomtee_object_private.h>
+
+/* MEMORY Object. */
+struct qcomtee_memmory {
```
Typo? qcomtee_memory?

qc-azarrabi replied:
Ack.

harshaldev27 commented on libqcomtee/src/objects/mem_obj.c:
```diff
@@ -8,10 +8,21 @@
 #include <qcomtee_object_private.h>
 
-/* MEMORY Object. */
+/* Which TEE API was used to prepare the meory object: */
```
Typo, 'memory object'.

qc-azarrabi replied:
Ack.

harshaldev27 commented on libqcomtee/src/qcomtee_object.c:
```diff
@@ -43,6 +45,8 @@ static int qcomtee_object_id_init(struct qcomtee_object *object,
 
 	/* Simple id allocator. */
 	for (i = 0; i < TABLE_SIZE; i++) {
+		if (idx == 0)
```
Comparing possibly uninitialized `idx` here is giving a compilation error..

qc-azarrabi replied:
Ack.

qc-azarrabi committed:
test: update ta_load test to use memory object

Run unittest path_to_bin 0 cmd [use buffer]
Run unittest path_to_bin 1 cmd [use memory object]

Signed-off-by: Amirreza Zarrabi <quic_azarrabi@quicinc.com>

harshaldev27 reviewed (approved):
This commit looks good to me.

harshaldev27 reviewed (approved):
This commit looks good to me.

harshaldev27 reviewed (commented):
This looks good to me.

harshaldev27 reviewed (commented):
This commit looks good to me.

harshaldev27 reviewed (commented):
This commit looks good to me.
